python/tests/scripts/b5_types_prelim.metta

```diff
@@ -13,97 +16,148 @@
   (Add S Z)
   S)
 
-; Adding type definitions. Arrow is a built-in symbol of the type system
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Adding type definitions.
+; Arrow (->) is a built-in symbol of the type system
 ; `Nat` is a custom symbol
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 (: Z Nat)
 (: S (-> Nat Nat))
 (: Add (-> Nat Nat Nat))
 ; Now the result of its evaluation will be a error expression, because `S` is not `Nat`
 !(assertEqualToResult
   (Add S Z)
   ((Error S BadType)))
-; This will still work, because `Something` is of undefined type,
-; and it might be `Nat`
-!(assertEqual
-  (Add Something Z)
-  Something)
-; This should work normally
+; But these other examples works normally
+!(assertEqualToResult
+  (Add (S Z) Z)
+  ((S Z)))
 !(assertEqual
   (Add Z (S Z))
   (S Z))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Adding to an untyped symbol (Something) will still work,
+; because it might be `Nat`
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+!(assertEqual
+  (Add Something Z)
+  Something)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ; Types can be nondeterministic
-; The following will be accepted by the interpreter
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 (: Ten Nat)
 (: Ten Int)
+; The following will be accepted by the interpreter
 !(assertEqualToResult
    (Add Z Ten)
   ((Add Z Ten)))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ; Types can be parameterized
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; List is defined as a dependent function type that
+; takes a Type and returns a Type
```

Comment:
This description sounds vague. List doesn't take `Type` as a symbol. Also, it is misleading to say that it returns a Type. Technically, it doesn't "return" anything.

Reply:
I guess I don't understand this completely... is this more accurate?
```
; List is defined as a function type with
; a typed argument
(: List (-> Type Type))
```

Reply:
To be accurate w.r.t. MeTTa semantics, I'd write something like "List has a function type. Its argument should be of Type type, and the type of the whole expression, e.g. (List String), will also have Type type." Does it sound too verbose or unclear?